tests/ui/impl-trait/precise-capturing/migration-note.stderr

```diff
@@ -0,0 +1,284 @@
+error[E0597]: `x` does not live long enough
+  --> $DIR/migration-note.rs:183:17
+   |
+LL |     let x = vec![0];
+   |         - binding `x` declared here
+LL |
+LL |     display_len(&x)
+   |     ------------^^-
+   |     |           |
+   |     |           borrowed value does not live long enough
+   |     argument requires that `x` is borrowed for `'static`
+...
+LL | }
+   | - `x` dropped here while still borrowed
+
+error[E0502]: cannot borrow `x` as mutable because it is also borrowed as immutable
+  --> $DIR/migration-note.rs:20:5
+   |
+LL |     let a = display_len(&x);
+   |                         -- immutable borrow occurs here
+...
+LL |     x.push(1);
+   |     ^^^^^^^^^ mutable borrow occurs here
+...
+LL |     println!("{a}");
+   |               --- immutable borrow later used here
+   |
+note: this call may capture more lifetimes since Rust 2024 has adjusted `impl Trait` lifetime capture rules
+  --> $DIR/migration-note.rs:17:13
+   |
+LL |     let a = display_len(&x);
+   |             ^^^^^^^^^^^^^^^
+help: add a `use<..>` precise capturing bound to avoid overcapturing
```

Comment:
I'd let the suggestion show the syntax and leave the text easier to read

```suggestion
help: add a precise capturing bound to avoid overcapturing in Rust 2024 edition
```

Reply:
I think saying "in Rust 2024" is redundant (that's the current edition enabled) and also going to be wrong as soon as we bump the edition further. I'll leave out the `use<..>` part tho.